pyccl/halos/profiles_2pt.py

```diff
@@ -106,3 +106,72 @@ def fourier_2pt(self, prof, cosmo, k, M, a,
                 raise ValueError("prof2 must be the same as prof")
 
         return prof._fourier_variance(cosmo, k, M, a, mass_def)
+
+
+class Profile2ptR(Profile2pt):
+    """ This class inherits from the 1-halo 2-point correlator between two
+    halo profiles. Sometimes we need not assume that the 1-halo terms are
+    fully correlated. We introduce a term `r_corr` which scales the product
+    of the fourier halo profiles by `(1+r_corr)`. In the trivial case where
+    `r_corr=0` this returns the 2-point correlator of the parent class.
+    Example usecases can be found in arXiv:1909.09102 and arXiv:2102.07701.
+    """
```

Comment:
If you're gonna make `r_corr` an input parameter, it needs to be defined in the docstring above.